bundles/org.openhab.binding.webexteams/src/main/java/org/openhab/binding/webexteams/internal/api/WebexTeamsApi.java

```diff
@@ -0,0 +1,178 @@
+@NonNullByDefault
+public class WebexTeamsApi {
+
+    private final Logger logger = LoggerFactory.getLogger(WebexTeamsApi.class);
+
+    private final @NonNullByDefault({}) OAuthClientService authService;
+    private final @NonNullByDefault({}) HttpClient httpClient;
+
+    /**
+     * Constructor.
+     * 
+     * @param WebexTeamsHandler hander interact with handler context.
+     */
+    public WebexTeamsApi(OAuthClientService authService, HttpClient httpClient) {
+        this.authService = authService;
+        this.httpClient = httpClient;
+    }
+
+    /**
+     * Get a <code>Person</code> object for the account.
+     * 
+     * @return a <code>Person</code> object
+     * @throws WebexAuthenticationException when authentication fails
+     * @throws WebexTeamsApiException for other failures
+     */
+    public Person getPerson() throws WebexTeamsApiException, WebexAuthenticationException {
+        URI url = getUrl(WEBEX_API_ENDPOINT + "/people/me");
+
+        Person person = request(url, HttpMethod.GET, Person.class, null);
+        return person;
+    }
+
+    private URI getUrl(@Nullable String url) throws WebexTeamsApiException {
+        URI uri;
+        try {
+            uri = new URI(url);
+        } catch (URISyntaxException e) {
+            throw new WebexTeamsApiException("bad url", e);
+        }
+        return uri;
+    }
+
+    private <I, O> O request(URI url, HttpMethod method, Class<O> clazz, I body)
+            throws WebexAuthenticationException, WebexTeamsApiException {
+        try {
+            // Refresh is handled automatically by this method
+            AccessTokenResponse response = this.authService.getAccessTokenResponse();
+
+            String authToken = response == null ? null : response.getAccessToken();
+            if (authToken == null) {
+                throw new WebexAuthenticationException("Auth token is null");
+            } else {
+                return doRequest(url, method, authToken, clazz, body);
+            }
+        } catch (OAuthException | IOException | OAuthResponseException e) {
+            throw new WebexAuthenticationException("Not authenticated", e);
+        }
+    }
+
+    private <I, O> O doRequest(URI url, HttpMethod method, String authToken, Class<O> clazz, I body)
+            throws WebexAuthenticationException, WebexTeamsApiException {
+        Gson gson = new Gson();
+        try {
+            Request req = httpClient.newRequest(url).method(method);
+            req.header("Authorization", "Bearer " + authToken);
+            logger.debug("Requesting {} with ({}, {})", url, clazz, body);
+
+            if (body != null) {
+                String bodyString = gson.toJson(body, body.getClass());
+                req.content(new StringContentProvider(bodyString));
+                req.header("Content-type", "application/json");
+            }
+
+            InputStreamResponseListener listener = new InputStreamResponseListener();
+            req.send(listener);
```

Comment:
A reason to prefer an asynchronous HTTP request ? 
With synchronous request, no need to consider aborting the request.

Reply:
Hmm.. no reason really.  I might have taken it from an other example.  I've changed to synchronous... might be a bit simpler.